aospy/utils/times.py

```diff
@@ -386,7 +386,36 @@ def ensure_time_avg_has_cf_metadata(ds):
             ds[coord].attrs['calendar'] = ds[TIME_STR].attrs['calendar']
 
     unit_interval = ds[TIME_STR].attrs['units'].split('since')[0].strip()
-    ds[AVERAGE_DT_STR].attrs['units'] = unit_interval
+    ds[TIME_WEIGHTS_STR].attrs['units'] = unit_interval
+    return ds
+
+
+def add_uniform_time_weights(ds):
+    """Append uniform time weights to a Dataset.
+
+    All DataArrays with a time coordinate require a time weights coordinate.
+    For Datasets read in without a time bounds coordinate or explicit
+    time weights built in, aospy adds uniform time weights at each point
+    in the time coordinate.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Input data
+
+    Returns
+    -------
+    Dataset
+    """
+    time = ds[internal_names.TIME_STR]
+    unit_interval = time.attrs['units'].split('since')[0].strip()
+    time_weights = xr.DataArray(
```

Comment:
FYI xarray has now implemented `ones_like` and similar helper functions: http://xarray.pydata.org/en/stable/generated/xarray.ones_like.html?highlight=ones_like

Might be slightly cleaner to use xarray's